### PR TITLE
Clarify why request's url field exists

### LINKED ISSUE #275: How does request url differ from url list and current url?
The spec says the following:

* A request has an associated url (a fetch URL). 
* A request has an associated url list (a list of one or more fetch URLs). Unless stated otherwise, it is a list containing a copy of request's url. 
* A request has an associated current url. It is a pointer to the last fetch URL in request's url list. 

Shouldn't the first bullet "associated url" just be the first URL in the list?  Why are we storing this separately?

### PR DESCRIPTION
**Pull request by annevk**
Fixes #275.

### CODE REVIEW

**Comment by annevk** [comment]
@wanderview, does this look reasonable?

File: Overview.html [hunk 1]
```diff
@@ -630,6 +630,11 @@ <h4 id="requests"><span class="secno">3.1.5 </span>Requests</h4>
 <p>A <a href="#concept-request" title="concept-request">request</a> has an associated
 <dfn id="concept-request-url" title="concept-request-url">url</dfn> (a <a href="#fetch-url">fetch URL</a>).
 
+<p class="note no-backref">Implementations are encouraged to make this a pointer for the first
```
Comment (wanderview) [inline Overview.html]: nit: "pointer *to* the first"

**Comment by wanderview** [comment]
Looks good.  Just a wording nit.  Thanks.

**Comment by annevk** [comment]
Thanks!

**Pull request merged**